tests/misc_test.go

```diff
@@ -535,3 +535,53 @@ func TestTrivialSwitch(t *testing.T) {
 	}
 	t.Fail()
 }
+
+func TestTupleFnReturnImplicitCast(t *testing.T) {
+	var ycalled int = 0
+	x := func(fn func() (int, error)) (interface{}, error) {
+		return fn()
+	}
+	y, _ := x(func() (int, error) {
+		ycalled++
+		return 14, nil
+	})
+	if y != 14 || ycalled != 1 {
+		t.Fail()
+	}
+}
+
+var tuple2called = 0
+
+func tuple1() (interface{}, error) {
+	return tuple2()
+}
+func tuple2() (int, error) {
+	tuple2called++
+	return 14, nil
+}
+func TestTupleReturnImplicitCast(t *testing.T) {
+	x, _ := tuple1()
+	if x != 14 || tuple2called != 1 {
+		t.Fail()
+	}
+}
+
+func TestDeferNamedTupleReturnImplicitCast(t *testing.T) {
```

Comment:
Hey @shurcooL,
I didn't actually check whether that test failed on the current master, I did however check the to see where the code I modified would actually be used form and I saw a special case when dealing with deferred parameters.  Additionally, the logic I modified was affected by named parameters as well.  I chose to put those two tests together and include it to ensure that case was tested, as the original singular test might have missed it.
tl;dr: Yea, entirely to catch future regressions from other people modifying the behaviour of the things surrounding my changes.
Cheers, Brett